src/cargo_communication/request_actor.rs

```diff
@@ -348,3 +348,837 @@ pub trait CargoHandleTrait<T> {
+                                },
+                            },
+                            message: "original diagnostic".to_string(),
+                        }]),
+                        tags: None,
+                        data: None,
+                    },
+                ],
+                reset: false,
+            },
+        ));
+
+        let proper_publish_diagnostic_second = Message::Notification(Notification::new(
+            PublishDiagnostics::METHOD.to_string(),
+            PublishDiagnosticsParams {
+                text_document: TextDocumentIdentifier {
+                    uri: "file:///test_root_path/test_filename2".into(),
+                },
+                build_target: "".into(),
+                //TODO change to "test_target_name" later
+                origin_id: Some("test_origin_id".into()),
+                diagnostics: vec![LSPDiagnostic {
+                    range: Range {
+                        start: Position {
+                            line: 0,
+                            character: 2,
+                        },
+                        end: Position {
+                            line: 1,
+                            character: 3,
+                        },
+                    },
+                    severity: Some(DiagnosticSeverity::HINT),
+                    code: Some(NumberOrString::String("test_code".to_string())),
+                    code_description: None,
+                    source: Some("cargo".into()),
+                    message: "test_label".to_string(),
+                    related_information: Some(vec![DiagnosticRelatedInformation {
+                        location: Location {
+                            uri: Url::parse("file:///test_root_path/test_filename1").unwrap(),
+                            range: Range {
+                                start: Position {
+                                    line: 0,
+                                    character: 2,
+                                },
+                                end: Position {
+                                    line: 1,
+                                    character: 3,
+                                },
+                            },
+                        },
+                        message: "original diagnostic".to_string(),
+                    }]),
+                    tags: None,
+                    data: None,
+                }],
+                reset: false,
+            },
+        ));
+
+        assert!(eq_unordered_vec(
+            &[
+                to_string(&proper_publish_diagnostic_first).unwrap(),
+                to_string(&proper_publish_diagnostic_second).unwrap(),
+            ],
+            &[
+                to_string(&recv_to_main.recv().unwrap()).unwrap(),
+                to_string(&recv_to_main.recv().unwrap()).unwrap()
+            ]
+        ));
+        assert!(recv_to_main
+            .recv_timeout(Duration::from_millis(100))
+            .is_err());
```

Comment:
to jest to "is_last_message" czy jakoś tak, co nie?

Reply:
Nie moge tutaj tego uzyc poniewaz ta linijka ma zapewnic ze juz nic nie zostanie nadeslane. Funkcja assert_is_last_msg gwarantuje ze nastepna odebrana wiadomosc bedzie ostatnią. Jednakze w wypadku tego testu nie wiemy ktora wiadomosc bedzie ostatnia z uwagi na losowosc hashowania w serwerze. Dlatego tez odbieram 2 wiadomosci i je sortuje leksykograficznie.